hedera-node/src/main/java/com/hedera/services/store/tokens/BaseTokenStore.java

```diff
@@ -460,11 +461,47 @@ protected ResponseCodeEnum tryAdjustment(AccountID aId, TokenID tId, long adjust
 			return INSUFFICIENT_TOKEN_BALANCE;
 		}
 		tokenRelsLedger.set(relationship, TOKEN_BALANCE, newBalance);
-		// TODO this should and will be updated in another PR
 		hederaLedger.updateTokenXfers(tId, aId, adjustment);
 		return OK;
 	}
 
+	protected ResponseCodeEnum tryAdjustment(AccountID senderAId, AccountID receiverAId, TokenID tId, long serialNumber) {
```

Comment:
- It seems to me that `tryAdjustment(AccountID aId, TokenID tId, long adjustment)` should be in the `common token store` since its for `FUNGIBLEs` and `tryAdjustment(AccountID senderAId, AccountID receiverAId, TokenID tId, long serialNumber)` is specifically for `unique token store` -> non fungibles.
Why don't we place them in their corresponding stores and have the shared logic (if any) in the base token store? Same thing goes for `sanityChecked` 


- It seems that the following code can be reused for both fungibles and non fungibles
```Java
if ((boolean) tokenRelsLedger.get(relationship, IS_FROZEN)) {
			return ACCOUNT_FROZEN_FOR_TOKEN;
} if (!(boolean) tokenRelsLedger.get(relationship, IS_KYC_GRANTED)) {
			return ACCOUNT_KYC_NOT_GRANTED_FOR_TOKEN;
}
```